### PR TITLE
Allow instrumentation of js code

### PR DESCRIPTION
**Pull request by evansiroky**
This adds the option to instrument code during browserify js builds.  This works in conjunction with the e2e tests performed in datatools-ui.

### CODE REVIEW

**Comment by evansiroky** [comment]
https://github.com/conveyal/datatools-ui/pull/394 is the PR where this would be used.

File: lib/babel-config.js [hunk 1]
```diff
@@ -20,6 +21,8 @@ module.exports = function (env) {
     reactRequire
   ]
 
+  if (env === 'instrumented') { plugins.push(istanbul) }
```
Comment (trevorgerhardt) [inline lib/babel-config.js]: Hmmm. Should this be it's own argument? Just `--instrument`.
 `env` is primarily used for test/dev/staging/production

**Comment by trevorgerhardt** [comment]
Hmmm. After seeing this set of changes it makes me wonder if it's possible to create the babel config higher up in the chain and pass that down instead of passing the `env` && `instrument` variables all the way through. What do you think @evansiroky?

**Comment by evansiroky** [comment]
I made this screenshot of the dependency graph.

screen shot 2019-01-14 at 3 43 25 pm

I think there are tradeoffs both ways.  If the babel config is created in the bin files, it would still have to go through the path `js-build` > `browserify` > `js-transform`.  Also, we'd have to send it from `mastarm-prepublish`.  I think with this method, just one variable is being passed through so maybe it's better than passing through a large object of stuff?

**Comment by trevorgerhardt** [comment]
Ah, interesting. Thanks for looking into that! Just keep it the way it is then.

**Pull request merged**